core/mcp.go

```diff
@@ -194,6 +194,109 @@ func (m *MCP) tools(srv *dagql.Server, typeName string) ([]LLMTool, error) {
 			// never a reason to call "sync" since we call it automatically
 			continue
 		}
+		// Hide functions from the largest and most commonly used core types,
+		// to prevent tool bloat
+		switch typeName {
+		case "Query":
+			switch field.Name {
+			case
+				"currentModule",
+				"currentTypeDefs",
+				"defaultPlatform",
+				"engine",
+				"env",
+				"error",
+				"function",
+				"generatedCode",
+				"llm",
+				"loadSecretFromName",
+				"module",
+				"moduleSource",
+				"secret",
+				"setSecret",
+				"sourceMap",
+				"typeDef",
+				"version":
+				continue
+			}
+		case "Container":
+			switch field.Name {
+			case
+				"build",
+				"defaultArgs",
+				"entrypoint",
+				"envVariable",
+				"envVariables",
+				"experimentalWithAllGPUs",
+				"experimentalWithGPU",
+				"export",
+				"exposedPorts",
+				"imageRef",
+				"import",
+				"label",
+				"labels",
+				"mounts",
+				"pipeline",
+				"platform",
+				"rootfs",
+				"terminal",
+				"up",
+				"user",
+				"withAnnotation",
+				"withDefaultTerminalCmd",
+				"withFile",
```

Comment:
nit: might as well keep this for the same reasoning as keeping `withDirectory` - it's common to encounter a `File`, and having to go through too many hops to turn it into `Directory` is less likely to work (let alone whether the model predicts that `withDirectory` copies additively)

Reply:
Done 👍